lib/src/Pos/Launcher/Configuration.hs

```diff
@@ -104,17 +109,35 @@ instance Default ConfigurationOptions where
 -- configuration at a given key.
 withConfigurations
     :: (WithLogger m, MonadThrow m, MonadIO m)
-    => ConfigurationOptions
+    => Maybe BlacklistPath
+    -> ConfigurationOptions
     -> (HasConfigurations => NtpConfiguration -> m r)
     -> m r
-withConfigurations co@ConfigurationOptions{..} act = do
-    logInfo ("using configurations: " <> show co)
-    Configuration{..} <- parseYamlConfig cfoFilePath cfoKey
-    let configurationDir = takeDirectory cfoFilePath
-    withCoreConfigurations ccCore configurationDir cfoSystemStart cfoSeed $
-        withUpdateConfiguration ccUpdate $
-        withSscConfiguration ccSsc $
-        withDlgConfiguration ccDlg $
-        withTxpConfiguration ccTxp $
-        withBlockConfiguration ccBlock $
-        withNodeConfiguration ccNode $ act ccNtp
+withConfigurations mBlacklistPath cfo act = do
+    logInfo ("using configurations: " <> show cfo)
+    cfg <- parseYamlConfig (cfoFilePath cfo) (cfoKey cfo)
+    -- Here is where we could read and add a blacklist.
+    blacklist <- case mBlacklistPath of
+        Nothing -> pure mempty
+        Just fp -> liftIO $ readBlacklistSrcAddrs fp
+    let configDir = takeDirectory $ cfoFilePath cfo
+    withCoreConfigurations (ccCore cfg) configDir (cfoSystemStart cfo) (cfoSeed cfo) $
+        withUpdateConfiguration (ccUpdate cfg) $
+        withSscConfiguration (ccSsc cfg) $
+        withDlgConfiguration (ccDlg cfg) $
+        withTxpConfiguration (addBlacklist blacklist $ ccTxp cfg) $
+        withBlockConfiguration (ccBlock cfg) $
+        withNodeConfiguration (ccNode cfg) $ act (ccNtp cfg)
```

Comment:
if you install editor-config, then it should pick up and respect the project-wide indentation settings :)

Reply:
Yeah, I'll have to look into editor-config. I'm one of those weirdos that doesn't use vim or emacs. I've been using geany for about 10 years now. Its not great, but i like it better than all the others editors I've tried in the last 10 years.